balancer/weightedroundrobin/weightedroundrobin.go

```diff
@@ -19,11 +19,38 @@
 // Package weightedroundrobin defines a weighted roundrobin balancer.
 package weightedroundrobin
 
+import (
+	"google.golang.org/grpc/resolver"
+)
+
 // Name is the name of weighted_round_robin balancer.
 const Name = "weighted_round_robin"
 
-// AddrInfo will be stored inside Address metadata in order to use weighted roundrobin
-// balancer.
+// attributeKey is the type used as the key to store AddrInfo in the Attributes
+// field of resolver.Address.
+type attributeKey struct{}
+
+// AddrInfo will be stored inside Address metadata in order to use weighted
+// roundrobin balancer.
 type AddrInfo struct {
 	Weight uint32
 }
+
+// SetAddrInfo returns a copy of addr in which the Attributes field is updated
+// with addrInfo.
+//
+// This is an EXPERIMENTAL API.
+func SetAddrInfo(addrInfo AddrInfo, addr resolver.Address) resolver.Address {
+	addr.Attributes = addr.Attributes.WithValues(attributeKey{}, addrInfo)
+	return addr
+}
+
+// GetAddrInfo returns the AddrInfo stored in the Attributes fields of addr.
+// Returns nil if no AddrInfo is present.
+//
+// This is an EXPERIMENTAL API.
+func GetAddrInfo(addr resolver.Address) AddrInfo {
+	v := addr.Attributes.Value(attributeKey{})
+	ai, _ := v.(AddrInfo)
```

Comment:
This panics if the info is not in attributes.

Reply:
Thanks. Done.

Reply:
Are you sure? I don't think it will.  The `, _` should prevent the panic when `v` is `nil`.

Reply:
https://play.golang.org/p/-DTj1hwQlPh

Reply:
[facepalm] Does it have `, _`? My bad.

Reply:
Wanted to add a test and make sure, but was lazy. Done now.